packages/ListBoxWithTags/src/ListBoxWithTags.js

```diff
@@ -220,13 +187,88 @@ export default function ListBoxWithTags(props) {
   );
 }
 
-ListBoxWithTags.propTypes = propTypes;
-ListBoxWithTags.defaultProps = defaultProps;
+ListBoxWithTags.displayName = "ListBoxWithTags";
 
+ListBoxWithTags.A11y = ListBox.A11y;
 ListBoxWithTags.Box = ListBox.Box;
 ListBoxWithTags.Divider = ListBox.Divider;
 ListBoxWithTags.Footer = ListBox.Footer;
 ListBoxWithTags.Option = ListBox.Option;
 ListBoxWithTags.Popover = ListBox.Popover;
 ListBoxWithTags.RawItem = ListBox.RawItem;
 ListBoxWithTags.filter = filter;
+
+ListBoxWithTags.types = {
+  size: {
+    MEDIUM: [constants.size.MEDIUM],
+    LARGE: [constants.size.LARGE],
+  },
+};
+
+ListBoxWithTags.propTypes = {
+  /** When this is true, it will display a message indicating all options are selected on the popover */
+  allOptionsAreSelected: PropTypes.bool,
+
+  /** Message to display when all options have been selected */
+  allOptionsAreSelectedMessage: PropTypes.string,
+
+  /** Child of type <ListBox.Option />, <ListBox.Divider />, etc */
+  children: PropTypes.arrayOf(PropTypes.node).isRequired,
+
+  /** Regex that match the input of the user and reports to onAddCustomOption. The default is a basic email regex */
+  customOptionRegex: PropTypes.instanceOf(RegExp),
+
+  /** filter function for the ListBoxWithTags can be pair with ListBoxWithTags.filter  */
+  filter: PropTypes.func,
+
+  /** If ListBox is in an error state  */
+  hasError: PropTypes.bool,
+
+  /** Disables the ListBox if true */
+  isDisabled: PropTypes.bool,
+
+  /** The ListBox will not allow value to be changed */
+  isReadOnly: PropTypes.bool,
+
+  /** String message to be display when there are not results  */
+  noResultsMessage: PropTypes.node,
+
+  /** Callback whenever the user input a new custom option like some@email.com, pass undefined to ignore this behaviour */
+  onAddCustomOption: PropTypes.func,
+
+  /** Callback whenever the user change a selection on the ListBoxWithTags  */
+  onChange: PropTypes.func,
+
+  /** Callback once a tag is remove from the Trigger */
+  onRemove: PropTypes.func,
+
+  /** Render prop to override the default Tag style, see example for it's uses.  */
+  renderTag: PropTypes.func,
+
+  /** An array of id that helps the ListBoxWithTags to known what elements are selected  */
+  selectedOptions: PropTypes.arrayOf(PropTypes.shape({})),
+
+  /** Size of the trigger and options (font size, height, padding, etc). */
+  size: PropTypes.oneOf([ListBoxWithTags.types.size.MEDIUM, ListBoxWithTags.types.size.LARGE]),
```

Comment:
After seeing `size` referenced repeatedly throughout the `ListBoxWithTags` source, I decided it made sense to include it in the explicit API for this component (even though all props for `ListBox` will get passed down to the underlying `<ListBox>`.

Reply:
I didn't put it as a prop because you then will have to include all listbox props which doesn't make sense is easier to say that the component inherent all listbox props which is what it does. 

then we don't have to expose a prop that doesn't give any benefit on the read me file.

Reply:
My opinion changed as I got into the code. Since the code of this component relies on that prop, and in fact manipulates it (for example, if `size="small"` is passed to this component, it will be adjusted to `size="medium"`) I think it should be an explicit part of the API.  

Additionally, prior to these changes, you could pass `isDisabled` to the `<ListBoxWithTags>` but it wouldn't have an effect (or maybe it only partially worked).  But now that it is handled explicitly in the code for this component, I think it's beneficial to include it in the PropTypes.  

It also means we don't need a bunch of `eslint-disable` statements, which makes the code smell better 😊

But I don't think it's necessary to include [all the ListBox props](https://github.com/acl-services/paprika/tree/master/packages/ListBox#props) just the ones that have some implementation details in this component.  And I've included [a note in the README](https://github.com/acl-services/paprika/pull/960/files#diff-18ef3a3c4a8b79a26448ca463dbc7de5af233afbe93ecb9d153584940b380f58R47-R53) to document the inclusion of the ListBox API.